Firestore/core/test/firebase/firestore/remote/serializer_test.cc

```diff
@@ -187,20 +188,30 @@ class SerializerTest : public ::testing::Test {
     *byte = new_value;
   }
 
+  v1::Value ValueProto(std::nullptr_t) {
+    ByteString bytes = EncodeFieldValue(FieldValue::Null());
+    return ProtobufParse<v1::Value>(bytes);
+  }
+
   v1::Value ValueProto(bool b) {
-    ByteString bytes =
-        EncodeFieldValue(&serializer, FieldValue::FromBoolean(b));
+    ByteString bytes = EncodeFieldValue(FieldValue::FromBoolean(b));
     return ProtobufParse<v1::Value>(bytes);
   }
 
   v1::Value ValueProto(int64_t i) {
-    ByteString bytes =
-        EncodeFieldValue(&serializer, FieldValue::FromInteger(i));
+    ByteString bytes = EncodeFieldValue(FieldValue::FromInteger(i));
     return ProtobufParse<v1::Value>(bytes);
   }
 
   v1::Value ValueProto(double d) {
-    ByteString bytes = EncodeFieldValue(&serializer, FieldValue::FromDouble(d));
+    ByteString bytes = EncodeFieldValue(FieldValue::FromDouble(d));
+    return ProtobufParse<v1::Value>(bytes);
+  }
+
+  // int64_t and double are equally good overloads for integer literals so this
+  // avoids ambiguity
+  v1::Value ValueProto(int i) {
+    ByteString bytes = EncodeFieldValue(FieldValue::FromInteger(i));
```

Comment:
Optional: just delegate to the `int64_t` overload? `return ValueProto(static_cast<int64_t>(i));`

Reply:
Done.